src/logics/Logic.cc

```diff
@@ -597,6 +586,9 @@ void Logic::simplifyTree(PTRef tr, PTRef& root_out)
 #endif
             for (int j = 0; j < t.size(); j++) {
                 PTRef cr = t[j];
+                if (isNonBoolIte(cr)) {
```

Comment:
Is this the expected behaviour? What happens in LALogic when there is an equality in the condition of a non-boolean ITE? It is not a problem that such equality would not be split to conjunction of inequalities?

Reply:
This splitting will be done on the output of the ITE manager system, on the constructed switches.